### PR TITLE
Add an error message if android NDK is not installed

### PR DESCRIPTION
**Pull request by ArchLinus**
I had trouble figuring out why I couldn't compile for android and realized I didn't have the right NDK version installed. So I figured I would add an error message to make it more obvious.

### CODE REVIEW

**Comment by dsnopek** [comment]
Does this actually lead to the message getting printed in your testing?

In looking into this previously, this code:

```
def exists(env):
    return get_android_ndk_root(env) is not None
```

... was preventing the `generate()` from running at all, and so any helpful message there wouldn't get shown (at least this was the case the last time I was looking into this problem).

See https://github.com/godotengine/godot-cpp/issues/791

**Comment by dsnopek** [comment]
Yeah, actually, if you look at the top of the `generate()` function, it already has a helpful error message, the problem is that `generate()` doesn't get run if the Android NDK isn't setup.

**Comment by ArchLinus** [comment]
> Does this actually lead to the message getting printed in your testing?

Yes, the printing worked, tested on Windows:

```
$ scons platform=android
scons: Reading SConscript files ...
Building for architecture arm64 on platform android
ERROR: Could not find NDK toolchain at C:/Users/<User>/AppData/Local/Android/Sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64.
Make sure NDK version 23.2.8568313 is installed.
```

whereas before I was just getting linker errors:

```
...
scons: *** [src\variant\vector4i.android.template_debug.arm64.o] The system cannot find the file specified
scons: *** [gen\src\variant\string.android.template_debug.arm64.o] The system cannot find the file specified        
scons: *** [gen\src\variant\string_name.android.template_debug.arm64.o] The system cannot find the file specified   
scons: building terminated because of errors.
```



> Yeah, actually, if you look at the top of the `generate()` function, it already has a helpful error message, the problem is that `generate()` doesn't get run if the Android NDK isn't setup.

For me it was getting run. The problem is that it doesn't actually check if the files exist on disk, all it does is construct a path to the NDK. I could modify it to do the exists check in `get_android_ndk_root` if you think that would be a better solution.

**Review by dsnopek — APPROVED** [review]
Ah, ok, so this is handling the case where `ANDROID_HOME` or `ANDROID_NDK_ROOT` are set, but NDK can't actually be found at the specified location?

That makes sense! And it seems to be working in my testing :-)

File: tools/android.py [hunk 1]
```diff
@@ -64,6 +64,12 @@ def generate(env):
     elif sys.platform == "darwin":
         toolchain += "darwin-x86_64"
         env.Append(LINKFLAGS=["-shared"])
+
+    if not os.path.exists(toolchain):
+        print("ERROR: Could not find NDK toolchain at " + toolchain + ".")
```
Comment (AThousandShips) [inline tools/android.py:69]:
```suggestion
        print("ERROR: Could not find NDK toolchain at " + toolchain + ".")
```
Trailing whitespace

Comment (ArchLinus) [reply tools/android.py]: Fixed

**Comment by dsnopek** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by dsnopek** [comment]
Cherry-picking to 4.2 in https://github.com/godotengine/godot-cpp/pull/1372

**Comment by dsnopek** [comment]
Cherry-picking to 4.1 in https://github.com/godotengine/godot-cpp/pull/1373

**Pull request merged**